### PR TITLE
feat(#933): D38-2.4a — backend REST + AIBlock rename (ADR-038 §6 Phase 3 backend)

### PR DESCRIPTION
**Pull request by zjzcpj**
Closes #933

## Summary

ADR-038 §6 Phase 3 backend deliverable: the four `/api/runs` REST endpoints, the markdown methods-export renderer, and the AIBlock `run_id` → `block_execution_id` rename per ADR-038 §5.2.

## Endpoints (ADR-038 §3.7, §3.8)

| Endpoint | Behavior |
|---|---|
| `GET /api/runs` | List runs (reverse-chrono) with optional `?workflow_id=` filter + `offset`/`limit` pagination. Returns `{runs, offset, limit, has_more}`. |
| `GET /api/runs/{run_id}` | Run row + joined `block_executions` ordered by `started_at`. |
| `GET /api/runs/{run_id}/methods` | Markdown body served as `text/markdown; charset=utf-8`. |
| `POST /api/runs/{run_id}/rerun` | Body `{execute_from_block_id?: str}`. Delegates to `ApiRuntime.start_workflow` so the new run gets recorded by the existing `LineageRecorder` path. Surfaces `ValueError` as 400, `FileNotFoundError` as 404. |

The `/_health` route from the D38-2.2 placeholder is preserved and now declared before `/{run_id}` so the literal segment matches first.

## Methods export

`src/scieasy/core/lineage/methods_export.py::render_methods_markdown` renders the four ADR-038 §3.7 user questions in one markdown body:

- **Q1**: Run identity + git commit + dirty marker + status + triggered_by + environment snapshot (JSON code-fence).
- **Q2**: Workflow YAML literal (yaml code-fence).
- **Q3**: Per-block `### {block_id} ({block_type} v{block_version})` subsections with timing + termination.
- **Q4a**: `block_config_resolved` per block (JSON code-fence).
- **Q4b**: Per-port I/O DataObjects as markdown tables, joining `block_io` ↔ `data_objects` (port name, position, type, storage path, object_id).

Renderer is read-only against `LineageStore`; missing-run returns a "Run not found" body rather than raising so the route can serve it inside any envelope it likes.

## AIBlock rename (ADR-038 §5.2)

- `RunDir.run_id` attribute + `__init__(project_dir, run_id)` parameter → `block_execution_id`.
- Path component `<project>/.scieasy/ai-block-runs/<run_id>/` → `<...>/<block_execution_id>/`.
- `AIBlock.run()` local variable + `_make_run_id` helper → `block_execution_id` + `_make_block_execution_id`.
- **Kept under legacy spelling** (external contracts): manifest JSON key `block.run_id` (ADR-035 §3.4 agent-facing schema) and `PtyTabSpec.block_run_id` field (ADR-034 engine freeze). Both have inline cross-reference docstrings explaining why.

## Tests

- `tests/api/test_runs_routes.py` (NEW, 20 tests): TestListRuns (×6), TestGetRun (×4), TestGetRunMethods (×4), TestRerunRun (×6 — uses `monkeypatch` to stub `ApiRuntime.start_workflow` so we test the routes layer, not the engine).
- `tests/core/test_methods_export.py` (NEW, 10 tests): happy-path Q1-Q4 + edge cases (unknown run, no blocks, empty YAML, dirty marker, block without I/O).
- `tests/blocks/ai/test_run_dir_skeleton.py` updated for the attribute rename.

All 794 of the relevant `tests/api/`, `tests/core/`, `tests/blocks/ai/` tests pass locally (1 pre-existing failure unrelated to this PR: `tests/api/test_projects.py::test_list_projects_prunes_deleted_directories` fails on `origin/track/adr-038/lineage-db` HEAD before this PR — missing optional plugin module). `ruff format --check` + `ruff check` clean on all changed files.

## Out of scope (per dispatch)

- ANY frontend file — D38-2.4b owns Lineage skeleton; D38-2.4c owns impl.
- `src/scieasy/core/lineage/store.py` / `record.py` / `recorder.py` (D38-2.2 merged).
- `src/scieasy/core/metadata_store.py` (D38-2.3 merged).
- ADR-039 owned files (different tracking branch).
- ADR doc files (only inline docstring cross-refs in `ai_block.py` + `run_dir.py`).

## Test plan

- [ ] CI green on `track/adr-038/lineage-db`.
- [ ] All four new endpoints exercised by `tests/api/test_runs_routes.py`.
- [ ] Methods renderer covered by `tests/core/test_methods_export.py`.
- [ ] AIBlock tests pass under new field name.
- [ ] Reconcile all Codex auto-review comments.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/scieasy/api/routes/runs.py [hunk 1]
```diff
@@ -1,39 +1,239 @@
...
+        # Defensive: a runs row without workflow_id is malformed, but we
+        # surface the failure as 422 so the client can distinguish from a
+        # plain "run not found".
+        raise HTTPException(
+            status_code=422,
+            detail=f"run {run_id!r} has no workflow_id; cannot determine target workflow",
+        )
+
+    try:
+        result = runtime.start_workflow(workflow_id, execute_from=body.execute_from_block_id)
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/runs.py:216]: accepted (will fix in next commit). Threading the source run_id through to ApiRuntime.start_workflow via a new parent_run_id kwarg so the new runs.parent_run_id column is populated. Also promoted triggered_by to ``rerun`` / ``rerun_from`` so the Lineage UI can render rerun provenance distinctly. New regression test ``TestRerunRun.test_threads_parent_run_id`` asserts the forward.
